cmd/datamon/cmd/flags.go

```diff
@@ -432,8 +435,21 @@ func paramsToDestStore(params flagsT,
 	return destStore, nil
 }
 
-func paramsToContributor(_ flagsT) (model.Contributor, error) {
-	return authorizer.Principal(config.Credential)
+func paramsToContributor(flags flagsT) (model.Contributor, error) {
```

Comment:
this appears to be the change in functionality that need land for v2 ..

..is it a workaround for
> some tests run os.Setenv(), which in turn pollute some other ones depending on the order of execution
or a change in the functional design of the program or both?

Reply:
work around. Also reflected on one of your (now famous) "???" in your latest PR regarding the env setup here. 
I don't think I really _changed_ the functionality... I made the influence of the env var more explicit and the priorities clearer.